autoload/sj/elm.vim

```diff
@@ -0,0 +1,247 @@
+" vim: foldmethod=marker
+
+" Split / Join {{{1
+"
+" function! sj#elm#Join() {{{2
+"
+" Attempts to join the multiline tuple, record or list closest
+" to current curspor position.
+function! sj#elm#Join()
+  let [from, to] = sj#elm#ClosestMultilineBraces()
+
+  if from[0] < 0
+    return 0
+  endif
+
+  let original = sj#GetByPosition([0, from[0], from[1]], [0, to[0], to[1]])
+  let joined = sj#elm#JoinOuterBraces(sj#elm#JoinLines(original))
+  call sj#ReplaceByPosition([0, from[0], from[1]], [0, to[0], to[1]], joined)
+
+  return 1
+endfunction
+
+" function! sj#elm#Split() {{{2
+"
+" Attempts to split the tuple, record or list fursest to current
+" curspor position on the same line.
+function! sj#elm#Split()
+  call sj#PushCursor()
+
+  let [from, to] = sj#elm#CurrentLineOutermostBraces(col('.'))
+
+  if from < 0
+    return 0
+  endif
+
+  let parts = sj#elm#SplitParts(from, to)
+
+  if len(parts) <= 2
+    return 0
+  endif
+
+  let replacement = join(parts, "\n")
+
+  call cursor(line('.'), from)
+  let [previousLine, previousCol] = searchpos('\S', 'bn')
+  if previousLine == line('.') && previousCol > 0 && sj#elm#CharAt(previousCol) =~ '[=:]'
+    let replacement = "\n".replacement
+    let from = previousCol + 1
+  end
+
+  call sj#ReplaceCols(from, to, replacement)
+
+  call sj#PopCursor()
+
+  return 1
+endfunction
+
+
+" Helper functions {{{1
+
+" function! sj#elm#SkipSyntax() {{{2
+"
+" Returns Elm's typical skippable syntax (strings and comments)
+function! sj#elm#SkipSyntax()
+  return sj#SkipSyntax(['elmString', 'elmTripleString', 'elmComment'])
+endfunction
```

Comment:
The syntax `sj#elm#Function()` uses namespacing that allows the function to be global while (hopefully) avoiding conflicts with other plugins. In a way, this turns this kind of function into "public API", and I try to mostly keep it consistent.

Helper functions don't need to be global, unless they're shared. For filetype-specific helper functions, I prefer to name them `s:FunctionName`. The `s:` prefix means "script-local" and it results in a function that can only be called in the current script. This can be a bit more annoying to test, because you can't call them in the console. But if you see a script-local function, that means it's safe to modify it without affecting other parts of the plugin, and it's safe to change them completely, because nobody else could be relying on them.

I would rename all the helper functions that are only used in this file with `s:` and the base name, without the namespacing.